autogpt_platform/frontend/src/components/analytics/google-analytics.tsx

```diff
@@ -0,0 +1,62 @@
+"use client";
+
+import { useEffect } from "react";
+import Script from "next/script";
+import type { GAParams } from "../../types/google";
+
+let currDataLayerName: string | undefined = undefined;
```

Comment:
Please make this not a global variable, or add a comment explaining why it has to be

Reply:
This is all pulled direct from next JS except loading the file location which is now local